lang/cpp/LC_DS.h

```diff
@@ -0,0 +1,14 @@
+struct ListNode {
```

Comment:
I was thinking moving these files into `testutils/cpp`, just like `testutils/go`.

Is it possible to generate a single cpp file like `1.two-sum.cpp` containing our generated `main` function, and includes the library from `$cppOutDir/common.h`.

We copies the library to `$cppOutdir/common.h` when `Initialize()`, and checks whether this file exists in `HasInitialized()`.

This way we can avoid generating the library code for every question, what do you think?

Reply:
I agree, that would be a cleaner solution. I will look into this later today.

It might even be worth it to compile this into an object file beforehand (with `Initialize()`) and link it during the compile phase. This should reduce compile time by quite a bit (it's like 1s / question currently on my machine, which isn't too great).

Reply:
> It might even be worth it to compile this into an object file beforehand (with `Initialize()`) and link it during the compile phase. This should reduce compile time by quite a bit (it's like 1s / question currently on my machine, which isn't too great).

If we do this, we might need to write some additional info into the workspace, like the version of compiler that compiled this library, so we can recompile the library when necessary. This way we can make sure it won't break when the compiler gets updated.

Reply:
Compiling into a object file can be a later optimization.